packages/webcomponents/src/ui-components/form/hidden-input.tsx

```diff
@@ -0,0 +1,63 @@
+import { Component, h, Host } from "@stencil/core";
+import { iframeInputStylesSet } from "./iframe-input-styles-state";
+
+@Component({
+  tag: "hidden-input",
+})
+export class HiddenInput {
+  private hiddenInput!: HTMLInputElement;
+
+  async componentDidLoad() {
+    const focusedStyles = await this.getFocusedStyles();
+    iframeInputStylesSet('focused', focusedStyles);
+
+    const getFocusedAndInvalidStyles = await this.getFocusedAndInvalidStyles();
+    iframeInputStylesSet('focusedAndInvalid', getFocusedAndInvalidStyles);
+  }
+
+  private async getFocusedStyles() {
+    return new Promise((resolve, _reject) => {
+      this.hiddenInput.focus();
+      setTimeout(() => {
+        this.hiddenInput.blur();
+        let computedStyles = getComputedStyle(this.hiddenInput);
+        resolve({
+          boxShadow: computedStyles.boxShadow,
+          border: computedStyles.border,
+        });
+      }, 1000);
+    });
+  }
+
+  private async getFocusedAndInvalidStyles() {
+    return new Promise((resolve, _reject) => {
+      this.hiddenInput.focus();
+      this.hiddenInput.classList.add('is-invalid');
+      setTimeout(() => {
+        let computedStyles = getComputedStyle(this.hiddenInput);
+        this.hiddenInput.blur();
+        this.hiddenInput.classList.remove('is-invalid');
+        const styles = {
+          boxShadow: computedStyles.boxShadow,
+          border: computedStyles.border,
+        };
+
+        resolve(styles);
+      }, 1000);
```

Comment:
And here? If this happens to be related to bootstrap animations, maybe we can remove the animation on the hidden input to get this as close to a 0 timeout as possible